### PR TITLE
Extract Abstractions to separate packages

### PR DESCRIPTION
**Pull request by TimHess**
Move abstractions to separate projects, review namespaces for consistency, improve package metadata, miscellaneous code cleanup #181,

### CODE REVIEW

File: src/Management/src/TracingBase/TracingOptions.cs [hunk 1]
```diff
@@ -81,17 +80,17 @@ internal TracingOptions()
...
-            {
-                CloudFoundryApplicationOptions appOptions = new CloudFoundryApplicationOptions(section);
-                if (!string.IsNullOrEmpty(appOptions.Name))
-                {
-                    return appOptions.Name;
-                }
-            }
-
-            section = config.GetSection(SPRING_APPLICATION_PREFIX);
+            // TODO: This functionality needs to be considered, but TracingBase shouldn't depend on CloudFoundryConfig
```
Comment (TimHess) [inline src/Management/src/TracingBase/TracingOptions.cs]: This needs attention

File: src/Management/src/EndpointCore/CloudFoundry/CloudFoundrySecurityMiddleware.cs [hunk 2]
```diff
@@ -126,24 +113,6 @@ private IEndpointOptions FindTargetEndpoint(PathString path)
         {
             List<IEndpointOptions> configEndpoints;
 
-            // Remove in 3.0
```
Comment (TimHess) [inline src/Management/src/EndpointCore/CloudFoundry/CloudFoundrySecurityMiddleware.cs]: @hananiel is this what is meant by "Remove in 3.0" or is it supposed to be replaced with something else?

File: src/Management/src/EndpointBase/CloudFoundry/CloudFoundryEndpoint.cs [hunk 3]
```diff
@@ -66,44 +53,41 @@ public override Links Invoke(string baseUrl)
...
-            else
-            {
-                var endpointOptions = Options.Global.EndpointOptions;
-                var links = new Links();
-
-                if (!Options.Enabled.Value)
-                {
-                    return links;
-                }
+            throw new NotImplementedException();
```
Comment (TimHess) [inline src/Management/src/EndpointBase/CloudFoundry/CloudFoundryEndpoint.cs]: @hananiel what should this do now?

File: src/CircuitBreaker/src/Hystrix.MetricsEventsCore/Steeltoe.CircuitBreaker.Hystrix.MetricsEventsCore.csproj [hunk 4]
```diff
@@ -1,22 +1,15 @@
...
-
+  <Import Project="..\..\..\..\versions.props" />
   <PropertyGroup>
-    <Description>Steeltoe Neflix Hystrix Metrics Event Stream ASP.NET Core</Description>
     <TargetFrameworks>netstandard2.0;netcoreapp3.0</TargetFrameworks>
-    <AssemblyName>Steeltoe.CircuitBreaker.Hystrix.MetricsEventsCore</AssemblyName>
-    <PackageId>Steeltoe.CircuitBreaker.Hystrix.MetricsEventsCore</PackageId>
-    <PackageTags>ASPNET Core;Circuit Breaker;Spring;Spring Cloud;Spring Cloud Hystrix;Hystrix</PackageTags>
+    <Description>Netflix Hystrix metrics event stream for ASP.NET Core over HTTP</Description>
+    <PackageTags>aspnetcoreore;Circuit Breaker;Spring;Spring Cloud;Spring Cloud Hystrix;Hystrix</PackageTags>
```
Comment (ddieruf) [inline src/CircuitBreaker/src/Hystrix.MetricsEventsCore/Steeltoe.CircuitBreaker.Hystrix.MetricsEventsCore.csproj]: aspnetcoreore?

File: src/CircuitBreaker/src/HystrixBase/Steeltoe.CircuitBreaker.HystrixBase.csproj [hunk 5]
```diff
@@ -1,26 +1,26 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <Import Project="..\..\..\..\versions.props" />
   <PropertyGroup>
-    <Description>Steeltoe Neflix Hystrix Client - Base Package</Description>
     <TargetFrameworks>netstandard2.0</TargetFrameworks>
-    <AssemblyName>Steeltoe.CircuitBreaker.HystrixBase</AssemblyName>
-    <PackageId>Steeltoe.CircuitBreaker.HystrixBase</PackageId>
+    <RootNamespace>Steeltoe.CircuitBreaker.Hystrix</RootNamespace>
+    <Description>An implementation of Neflix's Hystrix, for .NET</Description>
     <PackageTags>Spring Cloud;Netflix;Hystrix Client;Circuit Breaker</PackageTags>
```
Comment (ddieruf) [inline src/CircuitBreaker/src/HystrixBase/Steeltoe.CircuitBreaker.HystrixBase.csproj]: Add aspnetcore?

Comment (TimHess) [reply src/CircuitBreaker/src/HystrixBase/Steeltoe.CircuitBreaker.HystrixBase.csproj]: The base package doesn't have anything specific to asp.net core

File: src/Management/src/EndpointBase/CloudFoundry/CloudFoundryEndpoint.cs (hunk 3 above)
Comment (hananiel) [reply src/Management/src/EndpointBase/CloudFoundry/CloudFoundryEndpoint.cs]: @tim this should be removed as you did.

Comment (TimHess) [reply src/Management/src/EndpointBase/CloudFoundry/CloudFoundryEndpoint.cs]: Hold on - this now throws a `NotImplementedException`, that doesn't sound ideal

File: src/Common/src/Common/ApplicationInstanceInfo.cs [hunk 6]
```diff
@@ -44,7 +46,16 @@ public class ApplicationInstanceInfo : AbstractOptions, IApplicationInstanceInfo
 
         public string PlatformNameKey => BuildConfigString(PlatformRoot, ApplicationRoot + ":name");
 
-        protected static string PlatformRoot => string.Empty;
+        protected virtual string PlatformRoot => string.Empty;
+
+        protected void SecondChanceSetIdProperties(IConfiguration config = null)
```
Comment (hananiel) [inline src/Common/src/Common/ApplicationInstanceInfo.cs]: May be a comment on the constructors invoking this - rather than name it "second chance".

**Pull request merged**